src/MonitorProfiler/EventProvider/ProfilerEvent.h

```diff
@@ -111,10 +111,13 @@ template<typename... Args>
 template<size_t index, typename T, typename... TArgs>
 HRESULT ProfilerEvent<Args...>::WritePayload(COR_PRF_EVENT_DATA* data, const tstring& first, TArgs... rest)
 {
+    // Must be valid when next function is called
+    tstring empty;
```

Comment:
nit: Consider statically allocating this.

Reply:
And why does a new string object need to be used? Why can't `first` be used? Is that on the stack?